### PR TITLE
parametrized sources

### PR DESCRIPTION
**Pull request by fwbrasil**
@williamboxhall A macro would be easier do read, maintain and more flexible. Lets try without macros for now and see how it goes.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -12,6 +12,8 @@ libraryDependencies += "org.mockito" % "mockito-core" % "1.9.5" % "test"
 
 libraryDependencies += "com.twitter" %% "util-core" % "6.22.0"
 
+libraryDependencies += "org.scala-lang" % "scala-reflect" % scalaVersion.value
```
Comment (williamboxhall) [inline build.sbt]: What was this for?

Comment (fwbrasil) [inline build.sbt]: removed, it was necessary for the macro impl

**Comment by williamboxhall** [comment]
We should update the README too, and maybe do Zip.

It is a shame we have to use the `method _` notation everywhere but there is no way to use the overloaded `source` function without that :(. It was prettier when we didnt have to do that.

**Comment by fwbrasil** [comment]
Could you update the readme and try to change zip?

**Comment by williamboxhall** [comment]
Yeah, ok

**Comment by fwbrasil** [comment]
Thanks :)

**Comment by williamboxhall** [comment]
@fwbrasil how would you feel if i removed support for expanded functions on the way through? It makes things messier and it confuses IntelliJ (not that that is important, but just another reason)

**Comment by williamboxhall** [comment]
@fwbrasil I think this is ready to merge now

File: src/main/scala/io/getclump/Tuples.scala [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package io.getclump
+
+import com.twitter.util.Future
+
+trait Tuples {
+  def normalize1[A, B] = (inputs: (A, B)) => inputs match {
```
Comment (fwbrasil) [inline src/main/scala/io/getclump/Tuples.scala]: `protected`

Comment (williamboxhall) [inline src/main/scala/io/getclump/Tuples.scala]: same for the other traits I guess? how about apart from this?

**Comment by williamboxhall** [comment]
merged!

**Pull request merged**